core/src/main/scala/sttp/client3/logging/Log.scala

```diff
@@ -39,30 +39,64 @@ class DefaultLog[F[_]](
 ) extends Log[F] {
 
   def beforeRequestSend(request: Request[_, _]): F[Unit] =
+    request.loggingOptions match {
+      case Some(options) =>
+        before(
+          request,
+          options.logRequestBody.getOrElse(logRequestBody),
+          options.logRequestHeaders.getOrElse(logRequestHeaders)
+        )
+      case None => before(request, logRequestBody, logRequestHeaders)
+    }
+
+  private def before(request: Request[_, _], _logRequestBody: Boolean, _logRequestHeaders: Boolean): F[Unit] = {
     logger(
       beforeRequestSendLogLevel, {
         s"Sending request: ${
-            if (beforeCurlInsteadOfShow) request.toCurl
-            else request.show(includeBody = logRequestBody, logRequestHeaders, sensitiveHeaders)
+            if (beforeCurlInsteadOfShow && _logRequestBody && _logRequestHeaders) request.toCurl
+            else request.show(includeBody = _logRequestBody, _logRequestHeaders, sensitiveHeaders)
           }"
       }
     )
+  }
 
   override def response(
       request: Request[_, _],
       response: Response[_],
       responseBody: Option[String],
       elapsed: Option[Duration]
-  ): F[Unit] =
+  ): F[Unit] = request.loggingOptions match {
+    case Some(options) =>
+      toResponse(
+        request.showBasic,
+        response,
+        responseBody,
+        options.logResponseBody.getOrElse(responseBody.isDefined),
+        options.logResponseHeaders.getOrElse(logResponseHeaders),
+        elapsed
+      )
+    case None =>
+      toResponse(request.showBasic, response, responseBody, responseBody.isDefined, logResponseHeaders, elapsed)
+  }
+
+  private def toResponse(
```

Comment:
[minor] `toResponse` is misleading - we aren't converting anything to a response. Maybe `handleResponse` or simply `response` with a different set of parameters :)